types/wlr_screencopy_v1.c

```diff
@@ -269,18 +293,42 @@ static void frame_handle_copy(struct wl_client *wl_client,
 		return;
 	}
 
-	struct wl_shm_buffer *buffer = wl_shm_buffer_get(buffer_resource);
-	if (buffer == NULL) {
+	struct wlr_dmabuf_v1_buffer *dma_buffer = NULL;
+	struct wl_shm_buffer *shm_buffer = wl_shm_buffer_get(buffer_resource);
+	if (shm_buffer == NULL &&
+			wlr_dmabuf_v1_resource_is_buffer(buffer_resource)) {
+		dma_buffer =
+			wlr_dmabuf_v1_buffer_from_buffer_resource(buffer_resource);
+	}
+
+	if (shm_buffer == NULL && dma_buffer == NULL) {
```

Comment:
Hmm, so, the only way for a client to know whether a linux-dmabuf wl_buffer is supported by screencopy is to try and get a protocol error. Something more graceful would be nicer.

Reply:
As a client developer, that doesn't really bother me. The capability check would still need to be implemented on the client side and it wouldn't be much simpler than just trying to send a dmabuf before settling on an shm buffer if dmabuf fails the first time.

Reply:
This is a protocol error, this means that you get disconnected from the compositor. Also protocol errors should only be used for client bugs.

Reply:
Hmm, yes, I see. If a client tries to use dmabuf with an older compositor, they will get a protocol error. So, the only way to make things compatible is to extend the protocol.

Reply:
Right, bumping the protocol version is required here I think.

Reply:
Done